### PR TITLE
Do not rely on `USE_VST` as VST is a library now

### PR DESCRIPTION
**Pull request by crsib**
Resolves: *(direct link to the issue)*

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: libraries/lib-vst/VSTWrapper.h [hunk 1]
```diff
@@ -329,5 +327,3 @@ struct VSTMessage : EffectInstance::Message
 };
 
 #endif
-
-#endif // USE_VST
```
Comment (Paul-Licameli) [inline libraries/lib-vst/VSTWrapper.h:333]: Is this complete?  `USE_VST` occurs in 15 files.

Comment (crsib) [reply libraries/lib-vst/VSTWrapper.h:333]: I have force pushed the updated version.

**Comment by Paul-Licameli** [comment]
It's still incomplete.  USE_VST is still mentioned in src/AboutDialog.cpp and in src/CMakeLists.txt.

About the latter file -- my error, when I extracted lib-vst.  I made a mistake of not deleting lines no longer needed that refer to source files that no longer exist at paths under src/ .

**Comment by crsib** [comment]
I disagree. I have left it in `src/AboutDialog.cpp` as there is no other way to appropriately fill the dialog. However, `AboutDialog` does not rely on `target_compile_definitions`, instead it relies on `config*.h`, which is forcibly included in all translation units by `audacity_append_common_compiler_options`.

**Comment by Paul-Licameli** [comment]
> I disagree. I have left it in `src/AboutDialog.cpp` as there is no other way to appropriately fill the dialog. However, `AboutDialog` does not rely on `target_compile_definitions`, instead it relies on `config*.h`, which is forcibly included in all translation units by `audacity_append_common_compiler_options`.

OK, granted, but please clean up the nonexistent source file paths still in src/CMakeLists.txt, then I'll approve this.

**Comment by Paul-Licameli** [comment]
> I disagree. I have left it in `src/AboutDialog.cpp` as there is no other way to appropriately fill the dialog. However, `AboutDialog` does not rely on `target_compile_definitions`, instead it relies on `config*.h`, which is forcibly included in all translation units by `audacity_append_common_compiler_options`.

OK -- and I was confused about src/effects/VST, I didn't move all of it to libraries

**Review by Paul-Licameli — APPROVED** [review]

**Comment by Ryan-Miller-Audacity** [comment]
Tested.

Launch checks and quick VST2 sanity checks performed in MacOS, Windows, and Linux.

**Pull request merged**